builtin/providers/cloudstack/provider.go

```diff
@@ -1,30 +1,46 @@
 package cloudstack
 
 import (
+	"fmt"
+	"github.com/go-ini/ini"
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/terraform"
+	"os/user"
 )
 
 // Provider returns a terraform.ResourceProvider.
 func Provider() terraform.ResourceProvider {
+	usr, _ := user.Current()
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
+			"config": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  fmt.Sprintf("%s/.cloudmonkey/config", usr.HomeDir),
```

Comment:
You should not set a default here. Doing so will cause `providerConfigure` to always use the config file, even if you specified the API URL/key/secret.